app/controllers/api/v1/admin/site_settings_controller.rb

```diff
@@ -31,6 +32,22 @@ def update
 
           render_data status: :ok
         end
+
+        def destroy
+          site_setting = SiteSetting.joins(:setting)
+                                    .find_by(
+                                      provider: current_provider,
+                                      setting: { name: params[:name] }
+                                    )
+
+          return render_error status: :not_found unless site_setting
+
+          site_setting.image.purge if params[:name] == 'BrandingImage'
```

Comment:
We can check if the update operation succeeded,

Reply:
Isn't there a way to make this global? In application controller?
So if an update, destroy, create on a resource fails it returns a :bad_request?
Our methods are getting bulky with all the status returns.

Reply:
Of course it's feasible,
But, I don't think it's a good idea, it raises its own issues and complexity.
In the future there's a good chance that we'll be returning some information about `:bad_request` cases,